lib/schlepp/format/base.rb

```diff
@@ -70,9 +69,17 @@ def process!
       # Glob support to pull file list for processing.
       def retrieve_file_list
         if name.is_a? Array
-          name.flat_map { |n| Dir.glob(n) }
+          if Format.cwd == ''
```

Comment:
the only thing I can think of to make this simpler is to extract the cwd conditional into its own method, say get_file(filename). Then that method deals with the cwd conditional, and you deal with the Array or String conditional in retrieve_file_list.

Another thing you could do to remove the Array/String conditional is to just process name with Array(name). That will normalize the name to always be an array.

With those two things, you should be able to simplify this quite a bit.